Deprecate types in get field mapping API

mayya-sharipova opened the pull request:
- Add deprecation warning to RestGetFieldMappingAction
- Add two new java HRLC classes GetFieldMappingsRequest and
GetFieldMappingsResponse. These classes use new typeless forms
of a request and response, and differ in that from the server
versions.

Relates to #35190

elasticmachine commented:
Pinging @elastic/es-search

hub-cap commented on client/rest-high-level/src/main/java/org/elasticsearch/client/indices/GetFieldMappingsRequest.java:
```diff
@@ -0,0 +1,92 @@
+
+    private IndicesOptions indicesOptions = IndicesOptions.strictExpandOpen();
+
+    public GetFieldMappingsRequest() {}
```
is nothing actually required here in the request? If not then it makes sense to remove it

hub-cap commented on client/rest-high-level/src/test/java/org/elasticsearch/client/indices/GetFieldMappingsResponseTests.java:
```diff
@@ -0,0 +1,89 @@
+                final String mapping = randomBoolean() ? "{\"type\":\"string\"}" : "{\"type\":\"keyword\"}";
+                FieldMappingMetaData metaData =
+                    new FieldMappingMetaData("my field", new BytesArray(mapping));
+                fieldMappings.put("field" + k, metaData);
```
++ to this

mayya-sharipova replied:
@jtibshirani  Thanks for the comment. I have removed `FieldMappingMetaData.NULL` in the client `GetFieldMappingsResponse` as it is not necessary.  An empty `FieldMappingMetaData` will indicate that field(s) are not found.

mayya-sharipova commented:
@jtibshirani   and @hub-cap  thanks for the review. I have addressed all your comments.

@jtibshirani  This is ready for another review, whenever you have time.

jtibshirani commented on client/rest-high-level/src/main/java/org/elasticsearch/client/indices/GetFieldMappingsRequest.java:
```diff
@@ -0,0 +1,90 @@
+/** Request the mappings of specific fields */
+public class GetFieldMappingsRequest implements Validatable {
+
+    protected boolean local = false;
```
Small comment, can this be `private`?

jtibshirani commented on client/rest-high-level/src/test/java/org/elasticsearch/client/indices/GetFieldMappingsResponseTests.java:
```diff
@@ -0,0 +1,91 @@
+                int fields = randomInt(10);
+                for (int k = 0; k < fields; k++) {
+                    final String mapping = randomBoolean() ? "{\"type\":\"string\"}" : "{\"type\":\"keyword\"}";
+                    FieldMappingMetaData metaData = new FieldMappingMetaData("my field", new BytesArray(mapping));
```
Small comment, "my field" -> `randomAlphaOfLength(8)`.
